subxt/src/backend/mod.rs

```diff
@@ -336,54 +336,96 @@ pub struct StorageResponse {
+    pub use std::collections::{HashMap, VecDeque};
+    pub use subxt_core::{config::DefaultExtrinsicParams, Config};
+    pub use tokio::sync::{mpsc, Mutex};
+    pub type RpcResult<T> = Result<T, RpcError>;
+    pub type Item = RpcResult<String>;
+    pub use crate::backend::rpc::RawRpcSubscription;
+    pub use rpc::RpcClientT;
+    pub use serde_json::value::RawValue;
+
+    fn storage_response<K: Into<Vec<u8>>, V: Into<Vec<u8>>>(key: K, value: V) -> StorageResponse
+    where
+        Vec<u8>: From<K>,
+    {
+        StorageResponse {
+            key: key.into(),
+            value: value.into(),
+        }
+    }
+    pub mod rpc_client {
 
-    mod legacy {
-        use super::rpc::{RpcClient, RpcClientT};
-        use crate::backend::rpc::RawRpcSubscription;
-        use crate::backend::BackendExt;
-        use crate::{
-            backend::{
-                legacy::rpc_methods::Bytes, legacy::rpc_methods::RuntimeVersion,
-                legacy::LegacyBackend, StorageResponse,
-            },
-            error::RpcError,
-        };
-        use futures::StreamExt;
-        use serde::Serialize;
-        use serde_json::value::RawValue;
-        use std::{
-            collections::{HashMap, VecDeque},
-            sync::Arc,
-        };
-        use subxt_core::{config::DefaultExtrinsicParams, Config};
-        use tokio::sync::{mpsc, Mutex};
+        use std::time::Duration;
+
+        use super::*;
 
-        type RpcResult<T> = Result<T, RpcError>;
-        type Item = RpcResult<String>;
+        pub type SubscriptionHandler = Box<
+            dyn for<'a> Fn(
+                    &'a mut MockDataTable,
+                    &'a mut Option<Subscription>,
+                    Option<Box<serde_json::value::RawValue>>,
+                )
+                    -> super::rpc::RawRpcFuture<'a, super::rpc::RawRpcSubscription>
+                + Send,
+        >;
+
+        pub type MethodHandler = Box<
+            dyn for<'a> Fn(
+                    &'a mut MockDataTable,
+                    &'a mut Option<Subscription>,
+                    Option<Box<serde_json::value::RawValue>>,
+                )
+                    -> super::rpc::RawRpcFuture<'a, Box<serde_json::value::RawValue>>
+                + Send,
+        >;
 
-        struct MockDataTable {
-            items: HashMap<Vec<u8>, VecDeque<Item>>,
+        pub enum Message<T> {
+            Many(RpcResult<Vec<T>>),
+            Single(T),
         }
 
-        impl MockDataTable {
-            fn new() -> Self {
-                MockDataTable {
-                    items: HashMap::new(),
+        impl<T> Message<T> {
+            pub fn single(self) -> T {
+                match self {
+                    Self::Single(s) => s,
+                    _ => todo!(),
+                }
+            }
+            pub fn many(self) -> RpcResult<Vec<T>> {
+                match self {
+                    Self::Many(s) => s,
+                    _ => todo!(),
```

Comment:
Does something need doing here still or should it be a `panic!`?

Reply:
Replaced with panic